### ISSUE TITLE
HTTP/2 headers should all be lowercase

### ISSUE THREAD

**Issue by jwoertink** [luckyframework/lucky#1060]
According to the HTTP/2 spec, [all headers should be lowercase](https://http2.github.io/http2-spec/#rfc.section.8.1.2).

Currently crystal handles checking against lowercase vs uppercase, and dashes vs underscores, so `CONTENT_TYPE` is the same as `content-type`. We could safely lowercase, and dasherize for convention any place that we mention a header key directly. 

This also makes me curious to know what else we would need to do to ensure we fully support HTTP/2?  🤔

**Comment by wontruefree** [comment]
should this be a patch to the Crystal project?

**Comment by jwoertink** [comment]
Yeah, it probably should be. Does the HTTP::Server even support HTTP/2 yet?

**Comment by wontruefree** [comment]
I do not believe so.
but there is a reasonably well implementation here https://github.com/ysbaddaden/http2

**Comment by jwoertink** [comment]
With servers like [Caddy](https://github.com/caddyserver/caddy/releases/tag/v2.6.0) enabling HTTP/3 .... I wonder if there's stuff we would need to consider :thinking:  I know less than Jon Snow about it all
